fpga_interchange/fasm_generators/generic.py

```diff
@@ -141,7 +141,9 @@ def build_phys_cells_instances(self):
 
             bel = placement.bel
             bel_pins = placement.pins
-            cell_attr = self.logical_cells_instances.get(cell_name, None)
+
+            cell_attr = self.logical_cells_instances.get(
```

Comment:
this is alright for now, but makes me slightly uncomfortable as there is no reason a base cell name couldn't contain a `/`, for example via a Verilog escaped identifier.

my suggestion long term would be to modify the schema to include an explicit hierarchical path for cells rather than using a potentially ambiguous character.

Reply:
Indeed, this solution is a bit fragile. For context, this was required for the macro expansion case, where the original cell name (containing the attributes in the logical netlist) is enhanced with the various expanded cells (in the physical netlist).

I think that it should be possible to have some further checking whether the cell in question was expanded and, only in that case take the original name of the cell from the logical netlist

Reply:
I have added a check whether the cell is an expanded macro, even though it should become more robust in the future (probably with an addition to the physical netlist's schema as suggested).